lib/action.js

```diff
@@ -245,6 +245,28 @@ function action(system) {
 
 	});
 
+	// If a user wants to abort a single button actions
+	self.system.on('action_abort_one', function(bank) {
+
+		var bid = bank[0] + '_' + bank[1];
+		var b;
+		var c = 0;
+
+		while ((b = self.actions_running.indexOf(bid)) >= 0) {
+			if (c == 0) {
+				debug("Aborting button ",bank[0],",",bank[1]);
+			}
+			self.actions_running.splice(b,1);
```

Comment:
I think this is making a bad assumption about the relationship between `self.actions_running` and `self.timers_running`.

For each bank press with delayed actions, one entry will be added to actions_running on https://github.com/bitfocus/companion/blob/master/lib/action.js#L369, and multiple entries into timers_running https://github.com/bitfocus/companion/blob/master/lib/action.js#L409. The exact number depends on how many delayed actions there are.

For this to work, either the two arrays should be combined (perhaps using a Map of arrays?) or timers_running should contain pairs of values. The timer and the bank id.

Reply:
With a single delayed action, the actions<->timers arrays do map out. It works great to create a single-button press-repeat, cancel on release, without aborting everything else.
While working on a demonstration of how this works with multiple actions... As @Julusian pointed out it does not map properly with more than one delayed action. I was thinking about adding a bank/page id as part of the timers_running object to locate the timers involved.

Reply:
@istnv yeah I am currently thinking about a Map with keys being the timer, and the value being the bankid. You can then iterate over the map contents to find the items.
Im going to have a play with that quickly, and also refactor actions_running a bit to be more durable/simpler

Reply:
@istnv I have pushed a change to https://github.com/bitfocus/companion/tree/feat/abort_one that switches out these structures and some of the flow here to resolve my concerns.
Unless @krocheck has any concerns about my use of Map and Set instead of normal objects and arrays, I suggest using that as a fix for this comment thread

Reply:
@Julusian So I should close this PR and use your update as a base for the other changes suggested.

Reply:
You dont need to close the pr, you can pull the changes into your branch. 
`git fetch --all` then `git merge origin/feat/abort_one` (might not be called origin, depends on how your git remotes are setup) then continue making changes. there might be a conflict to resolve, but to resolve that you just need to make and commit the appropriate changes.

Reply:
I'm working against a fork, how do I pull a branch without rebuilding my repo?

Reply:
good point. make that
* `git remote add upstream https://github.com/bitfocus/companion.git` 
* `git fetch --all`
* `git merge upstream/feat/abort_one`

Reply:
Still learning git. Added a remote :)

Reply:
@Julusian if I knew what Map and Set were I'm sure I'd take issue for some reason, but since I don't I trust your judgement :)  You know I still suck at JS in certain areas.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

The same thing could be done with plain objects, but I generally prefer using these when possible. So its more of a style choice than anything